doc/user_guide/user_guide.md

```diff
@@ -1,4 +1,152 @@
 # User Guide
 
+This project contains a python library for describing Exasol error messages. 
+This library lets you define errors with a uniform set of attributes. 
+Furthermore, the error message  is implemented to be parseable, 
+so that you can extract an error catalog from the code.
 
-This user guide provides you with usage examples for this repository. 
\ No newline at end of file
+## Attributes of Error Builder
+
+Error messages are built in `ExaError` using following predefined attributes. 
+All attributes of the error object except the error code are optional. Thus, 
+after the error-code is provided, whether other attributes are provided or not 
+will not cause any error. Please keep in mind that error-code should satisfy 
```

Comment:
"whether other attributes are provided or not 
will not cause any error." reads weird. Maybe something like "not providing (some of) the other attributes will not raise an error"?